non-critical-infra/modules/backup-share.nix

```diff
@@ -0,0 +1,21 @@
+{ pkgs, ... }:
+{
+  environment.systemPackages = [ pkgs.cifs-utils ];
+  fileSystems."/mnt/backup-share" = {
+    device = "//u371748.your-storagebox.de/backup";
+    fsType = "cifs";
+    options =
+      let
+        # this line prevents hanging on network split
+        automount_opts = "x-systemd.automount,noauto,x-systemd.idle-timeout=60,x-systemd.device-timeout=5s,x-systemd.mount-timeout=5s";
+      in
+      [ "${automount_opts},credentials=/run/secrets/samba-storagebox" ];
+  };
```

Comment:
```suggestion
    options = [
      "credentials=/run/secrets/samba-storagebox"
      # the next lines are to prevent hanging on a network split
      "x-systemd.automount"
      "noauto"
      "x-systemd.idle-timeout=60"
      "x-systemd.device-timeout=5s"
      "x-systemd.mount-timeout=5s"
    ];
  };
```

Reply:
I think it should work that way and reads a bit clearer IMHO